book/src/internals/crate-organization.md

```diff
@@ -0,0 +1,157 @@
+<!--
+SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# Crate Organization
+
+CXX-Qt is made up of multiple crates under the `crates/` directory.
+
+This page lays out the goal of each crate and how it is differentiated from the other crates.
+
+- [⭐ `cxx-qt` - The Star of the Show](#cxx-qt)
+- [👷 `cxx-qt-build` - The Construction Worker](#cxx-qt-build)
+- [💚 `cxx-qt-gen` - The Heart of CXX-Qt](#cxx-qt-gen)
+- [📖 `cxx-qt-lib` - The "standard" Library](#cxx-qt-lib)
+- [📚 `cxx-qt-lib-extras` - The Bookshelf](#cxx-qt-lib-extras)
+- [💻 `cxx-qt-macro` - Frontend for cxx-qt-gen](#cxx-qt-macro)
+- [🦾 `qt-build-utils` - Building with Qt](#qt-build-utils)
+- [🤝 `cxx-qt-cmake` - Handoff to CMake](#cxx-qt-cmake)
+
+# <a name="cxx-qt"></a>⭐ `cxx-qt` - The Star of the Show
+
+The main crate of CXX-Qt.
+
+It serves as the actual crate our users should interface with.
+Anything that our users have to name in their downstream crates should be accessible via this crate.
+Mostly importantly this includes a re-export of the `#[cxx_qt::bridge]` macro from the `cxx-qt-macro` crate.
+Additionally, any code that the bridge references after expansion lives in `cxx-qt`.
+
+Limitation: The `cxx-qt` crate itself is limited to implementing a superset of CXX.
+This means its goal is to add additional Qt-specific features to CXX.
+Namely support for Qts Meta-Object-System, which includes QObjects, QEnums, properties, signals, slots, etc.
+
+CXX-Qt aims to **allow you** to wrap Qt APIs and create your own Qt-compatible types.
+It **does not** aim to actually wrap Qt API.
+Apart from a few extremely basic types that are needed for this goal, wrapping the actual Qt API is delegated to `cxx-qt-lib` and `cxx-qt-lib-extras`.
+
+This should also allow `cxx-qt-lib` and `cxx-qt-lib-extras` to be used without the user needing to use cxx-qt itself, which is why they're explicitly seperated.
```

Comment:
There is cxx-qt vs CXX-Qt, this sentence is using lowercase version compared to others using upper case

Reply:
Right, I'm mostly using the lower-case code-block version when referring to the specific crate.

And CXX-Qt to refer to the whole framework/concept.
I tried updating the PR to make this more consistent.